### ISSUE TITLE
Error Reading data from a BigQuery query - spark-3.1-bigquery-0.26.0-preview

### ISSUE THREAD

**Issue by AurelienLestienneNeoxia** [GoogleCloudDataproc/spark-bigquery-connector#722]
Hi,

I am aware this version of the connector is under development, still wanted to send out my bug report in case its any help. 
A NullPointerException in replaceCharVarcharWithStringInSchema arises when following the readme tutorial for reading SQL query results. 

(Tested in pyspark as well as in scala, with the same outcome)

Full Error stack: 
``` Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/spark/python/pyspark/sql/readwriter.py", line 204, in load
    return self._df(self._jreader.load(path))
  File "/usr/lib/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py", line 1304, in __call__
  File "/usr/lib/spark/python/pyspark/sql/utils.py", line 111, in deco
    return f(*a, **kw)
  File "/usr/lib/spark/python/lib/py4j-0.10.9-src.zip/py4j/protocol.py", line 326, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o73.load.
: java.lang.NullPointerException
        at org.apache.spark.sql.catalyst.util.CharVarcharUtils$.replaceCharVarcharWithStringInSchema(CharVarcharUtils.scala:41)
        at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:176)
        at org.apache.spark.sql.DataFrameReader.$anonfun$load$1(DataFrameReader.scala:303)
        at scala.Option.map(Option.scala:230)
        at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:265)
        at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:239)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:750)
```
Thanks for your work,

Environment:
jar package : gs://spark-lib/bigquery/spark-3.1-bigquery-0.26.0-preview
Dataproc image : "2.0.38-debian10"
Spark version 3.1.2
pyspark 3.1.2
Using Scala version 2.12.14, OpenJDK 64-Bit Server VM, 1.8.0_322

**Comment by davidrabinowitz** [comment]
Thanks for the report! We will research that

**Comment by suryasoma** [comment]
Hey @AurelienLestienneNeoxia, thanks for reporting the issue. The fix will be available in the next release.

**Comment by davidrabinowitz** [comment]
Fixed in spark-3.1-bigquery-0.27.0-preview
